sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -367,21 +433,51 @@ class GpuDynamicPartitionDataWriter(
         if (currentPartPath != partPath) {
           currentPartPath = partPath
           statsTrackers.foreach(_.newPartition())
-          fileCounter = 0
-          newOutputWriter(currentPartPath)
+
+          // see a new partition, close the old writer
+          val previousWriterStatus = currentWriterStatus
+          if(previousWriterStatus != null) {
```

Comment:
style nit: generally we have spaces after an `if` and before the opening `(`

Reply:
Done